### PR TITLE
Update the feature-state shortcode to display the default value.

### PR DESCRIPTION
**Pull request by vinayakankugoyal**
### Description

Update the feature-state shortcode to display the default value. This makes it clear what the feature-gate default value is and we don't need to update the docs content if we are only updating the feature-gate launch stage.

### Issue

Closes: #

Before change:
image

After change:
image

### CODE REVIEW

File: layouts/shortcodes/feature-state.html [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
             <div class="feature-state-notice feature-{{ .stage }}" title="{{ printf "%s %s" (T "feature_state_feature_gate_tooltip") $feature_gate_name }}">
               <span class="feature-state-name">{{ T "feature_state" }}</span> 
-              <code>{{ T "feature_state_kubernetes_label" }} v{{ .fromVersion }} [{{ .stage }}]</code>
+              <code>{{ T "feature_state_kubernetes_label" }} v{{ .fromVersion }} [{{ .stage }}] Default Value: {{.defaultValue}}</code>
```
Comment (dims) [inline layouts/shortcodes/feature-state.html:36]:
```suggestion
              <code>{{ T "feature_state_kubernetes_label" }} v{{ .fromVersion }} [{{ .stage }}] Enabled: {{.defaultValue}}</code>
```

File: layouts/shortcodes/feature-state.html [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
             <div class="feature-state-notice feature-{{ .stage }}" title="{{ printf "%s %s" (T "feature_state_feature_gate_tooltip") $feature_gate_name }}">
               <span class="feature-state-name">{{ T "feature_state" }}</span> 
-              <code>{{ T "feature_state_kubernetes_label" }} v{{ .fromVersion }} [{{ .stage }}]</code>
+              <code>{{ T "feature_state_kubernetes_label" }} v{{ .fromVersion }} [{{ .stage }}] Enabled: {{.defaultValue}}</code>
```
Comment (sftim) [inline layouts/shortcodes/feature-state.html:36]: We need to localize strings here.
I suggest two localized strings. The English versions could be:
`Enabled by default in Kubernetes {{ .version }}`
`Disabled by default in Kubernetes {{ .version }}`

and we could use a `<details>` element to allow revealing this plus other information. WDYT @vinayakankugoyal ?

See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details about the  `<details>` element.

---

Regardless, we _must_ support localizing the strings we use here.

Comment (vinayakankugoyal) [reply layouts/shortcodes/feature-state.html:36]: I would much prefer just showing the information inline, what other information is there to display?

How I would go about localizing this?

Comment (dipesh-rawat) [reply layouts/shortcodes/feature-state.html:36]: > How I would go about localizing this?

@vinayakankugoyal To localize the hardcoded string "Enabled" _(or any other string)_, simply add it to the **[en.toml](https://github.com/kubernetes/website/blob/main/data/i18n/en/en.toml)** file. Then, utilize the Hugo translate  'T' [function](https://gohugo.io/functions/lang/translate/) to fetch its value.  

Check out how [`feature_state_kubernetes_label`](https://github.com/kubernetes/website/blob/2f018d48603466c2a9a4a7891dda8cb581669128/data/i18n/en/en.toml#L200) has been used in the same file for guidance.

**Comment by dipesh-rawat** [comment]
Nice improvement. Could we consider targeting this PR to the _main_ branch instead of the _dev -1.32_ branch?

File: layouts/shortcodes/feature-state.html (hunk 2 above)
Comment (vinayakankugoyal) [reply layouts/shortcodes/feature-state.html]: Updated. Thanks for the pointers!

**Comment by vinayakankugoyal** [comment]
> Nice improvement. Could we consider targeting this PR to the _main_ branch instead of the _dev -1.32_ branch?

As long as we cherrypick it into dev-1.32

File: layouts/shortcodes/feature-state.html [hunk 3]
```diff
@@ -33,7 +33,7 @@
 
             <div class="feature-state-notice feature-{{ .stage }}" title="{{ printf "%s %s" (T "feature_state_feature_gate_tooltip") $feature_gate_name }}">
               <span class="feature-state-name">{{ T "feature_state" }}</span> 
-              <code>{{ T "feature_state_kubernetes_label" }} v{{ .fromVersion }} [{{ .stage }}]</code>
+              <code>{{ T "feature_state_kubernetes_label" }} v{{ .fromVersion }} [{{ .stage }}] {{ T "feature_gate_enabled" }}: {{.defaultValue}}</code>
```
Comment (sftim) [inline layouts/shortcodes/feature-state.html:36]: (nit) Move the colon symbol into the localized string.
Or try this:
```suggestion
              <code>{{ T "feature_state_kubernetes_label" }} v{{ .fromVersion }} [{{ .stage }}]</code> {{ T "feature_gate_enabled" (dict "enabled" .defaultValue) }}
```
and make a matching change to the localized string, such as
```toml
other = "(enabled by default: {{ .enabled }})"
```

Comment (vinayakankugoyal) [reply layouts/shortcodes/feature-state.html]: Done. Thanks!

**Comment by sftim** [comment]
Preview at https://deploy-preview-48422--kubernetes-io-main-staging.netlify.app/docs/concepts/overview/kubernetes-api/#aggregated-discovery looks good

/lgtm

**Comment by sftim** [comment]
One side comment. Updating the docs is a bit of a forcing function - I quite like that when people promote a feature to beta, they also have to at least glance at the docs for the thing they are promoting.

Getting reviewer eyes there is helpful to, to my mind.

**Comment by natalisucks** [comment]
thanks a bunch @vinayakankugoyal!
/approve

**Pull request merged**